lib/generate-critical-css.js

```diff
@@ -189,10 +189,15 @@ async function generateCriticalCSS( {
 		// Collect warnings / errors together.
 		const warnings = cssFiles.getErrors().concat( cssErrors );
 
-		return [ css, warnings ];
+		return [ cleanupCSS( css ), warnings ];
 	} finally {
 		browserInterface.cleanup();
 	}
 }
 
+function cleanupCSS( css ) {
+	// Remove keyframe animations.
+	return css.replace( /@[\w-]*?keyframes\s+[\w-]*\s*\{.*?\}\s*\}/gis, '' );
```

Comment:
I don't think we should use regexps to remove keyframes; we should do it in the AST as it will be faster and less prone to surprise corner-cases in the syntax.

Take a look at the `prune*` methods in `lib/style-ast.js` for examples on how to do it.

Reply:
Thanks for the hint. I have changed accordingly. I have also excluded `animation` rules from css.